src/rosdiscover/config.py

```diff
@@ -96,12 +99,19 @@ def from_dict(cls, dict_: t.Mapping[str, t.Any]) -> 'NodeSourceInfo':
                 raise ValueError("expected 'entrypoint' to be a string")
             entrypoint = dict_['entrypoint']
 
+        restricted_paths = []
+        if 'restrict-analysis-to-paths' in dict_:
+            if not isinstance(dict_['restrict-analysis-to-paths'], list):
+                raise ValueError("expected 'restrict-analysis-to-paths' to bae a list")
+            restricted_paths = dict_['restrict-analysis-to-paths']
```

Comment:
very minor suggestion: you can just define `restricted_paths = dict_.get("restrict-analysis-to-paths", [])` after you've done the sanity checking above.